internal/worker/client.go

```diff
@@ -344,14 +355,8 @@ func (c *Client) RequestJob(types []string, arch string) (Job, error) {
 		Arch:  arch,
 	}
 
-	workerID := func() uuid.UUID {
-		c.workerIDMu.RLock()
-		defer c.workerIDMu.RUnlock()
-		return c.workerID
-	}()
-
-	if workerID != uuid.Nil {
-		reqBody.WorkerId = common.ToPtr(workerID.String())
+	if c.getWorkerID() != uuid.Nil {
```

Comment:
`getWorkedIDLocked` sounds to me like there could be another version without a lock - but there shouldn't be any (without a lock) but it could be renamed if that makes it more clear

Reply:
… and yes - it's kind of racy - I'll change that

Reply:
This is still racy, assign it to a variable first.